### ISSUE TITLE
Removing Categories through REST API

### ISSUE THREAD

**Issue by waqasakramdot** [dotCMS/core#22756]
**Describe the bug**
Whenever REST API call made to clear/remove categories it returns 500

**To Reproduce**
Example. 

News has category **type** so Today news has two categories sport, entertainment so when trying to remove all categories with **type:{}** It returns 500
/api/v1/workflow/actions/fire
{ 
   "actionName": "Save",
   "comments": "saving content",
   "contentlet": {
      "contentType":"News",
        "identifier": "f3fd1b2042418d84592334b3b9a27627",
        "inode": "22540f01-7322-4c79-bb1a-e57810a85f0b",
        "type":{},
      "title":"blkjjhj Second Content", 
      "languageId": "1"
  }
}.

Logs:

`ERROR workflow.WorkflowResource: Exception on firing, workflow action: ceca71a0-deee-4999-bd47-b01baa1bcfc8, inode: null
java.lang.NullPointerException: null
at com.dotmarketing.portlets.categories.business.CategoryAPIImpl.findByVariable_aroundBody72(CategoryAPIImpl.java:787) ~[dotcms_22.03.2_999999.jar:?]
at com.dotmarketing.portlets.categories.business.CategoryAPIImpl$AjcClosure73.run(CategoryAPIImpl.java:1) ~[dotcms_22.03.2_999999.jar:?]
at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.9.2.jar:?]`

**Comment by wezell** [comment]
Just to be clear - the pattern for categories and relationships is this:
- pass a list of values, replace existing categories with these values
- pass null or ignore the property, then maintain current categories
- pass an empty list [] - then you remove all the categories

**Comment by nollymar** [comment]
PR: #22820

**Comment by nollymar** [comment]
**Internal QA:** Needs work. Docker image used: `dotcms/dotcms:22.09_60ad81cf_SNAPSHOT`

I tested the following escenarios: 

Content Type definition: 
<img width="1489" alt="Screen Shot 2022-08-31 at 11 26 14 AM" src="https://user-images.githubusercontent.com/8741395/187730141-09838c9e-2380-4f58-9dcc-5e3efa1b2836.png">

1. Save a new piece of content without categories using Workflow API. Test Result: OK.
```
PUT localhost:8080/api/v1/workflow/actions/fire
{
    "actionName": "save",
    "comments": "saving content",
    "contentlet": {
      "contentType":"MyType",
      "title": "Content 2",
      "mycats": [], 
    }
} 
```

2. Update the piece of content adding new categories using Workflow API. Test Result: OK.
```
PUT localhost:8080/api/v1/workflow/actions/fire
{
    "actionName": "save",
    "comments": "saving content",
    "contentlet": {
      "contentType":"MyType",
      "title": "Content 2",
      "mycats": ["baby", "toddler", "backcountry"], 
      "identifier": "8f2eecb0211e75325b6da6644427700d",
      "inode": "379bc15a-1564-4c2d-ba35-0fdba53db9e2"
    }
}  
```

3. Update the piece of content sending `null` categories using Workflow API. Test Result: Categories are wiped out when they should have been ignored.
```
PUT localhost:8080/api/v1/workflow/actions/fire
{
    "actionName": "save",
    "comments": "saving content",
    "contentlet": {
      "contentType":"MyType",
      "title": "Content 2",
      "mycats": null, 
      "identifier": "8f2eecb0211e75325b6da6644427700d",
      "inode": "379bc15a-1564-4c2d-ba35-0fdba53db9e2"
    }
}   
```
4. Update the piece of content ignoring the category field using Workflow API. Test Result: Categories are wiped out when they should have been ignored.
```
PUT localhost:8080/api/v1/workflow/actions/fire
{
    "actionName": "save",
    "comments": "saving content",
    "contentlet": {
      "contentType":"MyType",
      "title": "Content 2"
      "identifier": "8f2eecb0211e75325b6da6644427700d",
      "inode": "379bc15a-1564-4c2d-ba35-0fdba53db9e2"
    }
}   
```

**Comment by fabrizzio-dotCMS** [comment]
new PR https://github.com/dotCMS/core/pull/22930/

**Comment by nollymar** [comment]
**Internal QA:** Passed

All the test cases are working correctly, even with multiple category fields.

**Comment by josemejias07** [comment]
**QA Comment**

I went ahead and tested the 8 points using mentioned by Nollymar using postman. I gave my report to Bryan and he will be finishing the work on this.

**Comment by bryanboza** [comment]
According to the merged changes, we add a postman test just for the `workflow resource`, but we made changes in the `content resource` too, then we need to add a postman test to those changes.

**Comment by fabrizzio-dotCMS** [comment]
@bryanboza if you take the time to review the PR associated with this fix you'll notice that there is a new Postman test already there.

**Comment by nollymar** [comment]
Postman test implemented here: #22966 22966

**Comment by bryanboza** [comment]
Fixed, tested on release-22.09 // Docker // Postman
